ts/adaptors/linkedomAdaptor.ts

```diff
@@ -0,0 +1,148 @@
+
+/**
+ * @fileoverview  Implements the linkedom DOM adaptor
+ *
+ * @author dpvc@mathjax.org (Davide Cervone)
+ */
+
+import {HTMLAdaptor} from './HTMLAdaptor.js';
+import {userOptions, defaultOptions, OptionList} from '../util/Options.js';
+
+export class LinkedomAdaptor extends HTMLAdaptor<HTMLElement, Text, Document> {
+
+  /**
+   * The default options
+   */
+  public static OPTIONS: OptionList = {
+    fontSize: 16,          // We can't compute the font size, so always use this
+    fontFamily: 'Times',   // We can't compute the font family, so always use this
+    cjkCharWidth: 1,       // Width (in em units) of full width characters
+    unknownCharWidth: .6,  // Width (in em units) of unknown (non-full-width) characters
+    unknownCharHeight: .8, // Height (in em units) of unknown characters
+  };
+
+  /**
+   * Pattern to identify CJK (i.e., full-width) characters
+   */
+  public static cjkPattern = new RegExp([
+    '[',
+    '\u1100-\u115F', // Hangul Jamo
+    '\u2329\u232A',  // LEFT-POINTING ANGLE BRACKET, RIGHT-POINTING ANGLE BRACKET
+    '\u2E80-\u303E', // CJK Radicals Supplement ... CJK Symbols and Punctuation
+    '\u3040-\u3247', // Hiragana ... Enclosed CJK Letters and Months
+    '\u3250-\u4DBF', // Enclosed CJK Letters and Months ... CJK Unified Ideographs Extension A
+    '\u4E00-\uA4C6', // CJK Unified Ideographs ... Yi Radicals
+    '\uA960-\uA97C', // Hangul Jamo Extended-A
+    '\uAC00-\uD7A3', // Hangul Syllables
+    '\uF900-\uFAFF', // CJK Compatibility Ideographs
+    '\uFE10-\uFE19', // Vertical Forms
+    '\uFE30-\uFE6B', // CJK Compatibility Forms ... Small Form Variants
+    '\uFF01-\uFF60\uFFE0-\uFFE6', // Halfwidth and Fullwidth Forms
+    '\u{1B000}-\u{1B001}', // Kana Supplement
+    '\u{1F200}-\u{1F251}', // Enclosed Ideographic Supplement
+    '\u{20000}-\u{3FFFD}', // CJK Unified Ideographs Extension B ... Tertiary Ideographic Plane
+    ']'
+  ].join(''), 'gu');
+
+  /**
+   * The options for the instance
+   */
+  public options: OptionList;
+
+  /**
+   * @param {Window} window   The window to work with
+   * @param {OptionList} options  The options for the linkedom adaptor
+   * @constructor
+   */
+  constructor(window: Window, options: OptionList = null) {
+    super(window);
+    window.constructor.prototype.HTMLCollection = class {};  // add fake class for missing HTMLCollecton
+    let CLASS = this.constructor as typeof LinkedomAdaptor;
+    this.options = userOptions(defaultOptions({}, CLASS.OPTIONS), options);
+  }
+
+  /**
+   * @override
+   */
+  public parse(text: string, format: string = 'text/html') {
+    if (!text.match(/^(?:\s|\n)*</)) text = '<html>' + text + '</html>';
+    return this.parser.parseFromString(text, format);
+  }
+
+  /**
+   * @override
+   *
+   * This will do an HTML serialization, which may be good enough, but
+   *   won't necessarily close some tags properly.
+   */
+  public serializeXML(node: HTMLElement) {
+    return this.outerHTML(node);
+  }
+
+  /**
```

Comment:
Why not use a pollyfill for getComputedStyle?

E.g. see https://gist.github.com/abbotto/19a6680bf052e8c64f6e and https://github.com/mikemadest/jest-environment-linkedom/blob/main/src/get-computed-style-polyfill.js

Reply:
(Sorry, see my comment below.  I entered it in the wrong box.)